compiler-core/src/codegen.rs

```diff
@@ -242,8 +257,30 @@ impl<'a> JavaScript<'a> {
             &module.code,
             self.target_support,
             self.typescript,
+            &mut source_map_emitter,
         );
         tracing::debug!(name = ?js_name, "Generated js module");
-        writer.write(&path, &output?)
+        writer.write(&path, &output?)?;
+
+        match source_map_emitter {
+            SourceMapEmitter::Null => Ok(()),
+            SourceMapEmitter::Emit(builder) => {
```

Comment:
Extract this clause to a helper module please

Reply:
Could you clarify what is a helper module ? Do you mean that this code belongs in `sourcemap.rs` ?